tests/integration/test_charm.py

```diff
@@ -0,0 +1,409 @@
+        ops_test.model.applications[PRODUCER].units[0],
+        "start-process",
+        producer_credentials["kafka"]["endpoints"],
+        producer_credentials["kafka"]["username"],
+        producer_credentials["kafka"]["password"],
+        producer_credentials["kafka"]["topic"],
+    )
+
+    logger.info(f"Res: {res_producer}")
+
+    res_consumer = await fetch_action_start_process(
+        ops_test.model.applications[CONSUMER].units[0],
+        "start-process",
+        consumer_credentials["kafka"]["endpoints"],
+        consumer_credentials["kafka"]["username"],
+        consumer_credentials["kafka"]["password"],
+        consumer_credentials["kafka"]["topic"],
+        consumer_group_prefix=consumer_credentials["kafka"]["consumer-group-prefix"],
+    )
+
+    logger.info(f"Res: {res_consumer}")
+
+    time.sleep(20)
+
+    res_stop_producer = await fetch_action_stop_process(
+        ops_test.model.applications[PRODUCER].units[0],
+        "stop-process",
+    )
+    logger.info(f"Stop result producer: {res_stop_producer}")
+    res_stop_consumer = await fetch_action_stop_process(
+        ops_test.model.applications[CONSUMER].units[0],
+        "stop-process",
+    )
+    logger.info(f"Stop result consumer: {res_stop_consumer}")
+    await ops_test.model.remove_application(DATA_INTEGRATOR_CONSUMER)
+    await ops_test.model.remove_application(DATA_INTEGRATOR_PRODUCER)
+
+    await ops_test.model.wait_for_idle(apps=[KAFKA])
+
+
+@pytest.mark.abort_on_fail
+async def test_deploy_mongodb_and_relate(ops_test: OpsTest, kafka_app_charm):
+    """Deploy mongoDB, relate it with the kafka-app and dump messages."""
+    # clean topic
+
+    consumer_config = {"role": "consumer", "num_messages": "20", "topic_name": "topic_1"}
+    producer_config = {"role": "producer", "num_messages": "20", "topic_name": "topic_1"}
+
+    await ops_test.model.applications[PRODUCER].set_config(producer_config)
+    await ops_test.model.wait_for_idle(apps=[PRODUCER])
+
+    await ops_test.model.applications[CONSUMER].set_config(consumer_config)
+    await ops_test.model.wait_for_idle(apps=[CONSUMER])
+
+    await asyncio.gather(
+        ops_test.model.deploy(
+            MONGODB,
+            channel="5/edge",
+            application_name=MONGODB,
+            num_units=1,
+            series="jammy",
+        ),
+    )
+    await ops_test.model.wait_for_idle(apps=[MONGODB], timeout=1000, status="active")
+    await ops_test.model.add_relation(MONGODB, PRODUCER)
+    await ops_test.model.wait_for_idle(apps=[MONGODB, PRODUCER])
+    await ops_test.model.add_relation(MONGODB, CONSUMER)
+    await ops_test.model.wait_for_idle(apps=[MONGODB, CONSUMER])
+
+    # write messages to MongoDB
+    await ops_test.model.add_relation(KAFKA, PRODUCER)
+    await ops_test.model.wait_for_idle(apps=[KAFKA, PRODUCER])
+
+    await ops_test.model.add_relation(KAFKA, CONSUMER)
+    await ops_test.model.wait_for_idle(apps=[KAFKA, CONSUMER])
+
+    time.sleep(60)
```

Comment:
**todo:** I think the reason you might need to `time.sleep()` here is because when we `add_relation` and Python starts processing `wait_for_idle`, Juju is basically 'waking up' and everything is still 'idle', so it passes and moves on to the next.

Solve it with `idle_period=10` in the `wait_for_idle`s

Reply:
By the way, we could also use the `ops.fast_forward()` to increate the frequency with which "update_status" event are fired